### PR TITLE
HYDRAN-135 Implement Snailmail integration

### PR DESCRIPTION
**Pull request by AndreSoderlund**
* Implemented the calls to messaging for sending snail mails. 
* Added semaphore to cap the number of active threads.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [x] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [x] Yes (I have stepped the version number accordingly)
- [ ] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/integration-test/resources/MessageLetterIT/__files/test02_successfully_sendSnailMail/mappings/api-messaging-send-snail-mail-201.json [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {
 	"request": {
-		"url": "/api-messaging/2281/snail-mail",
-		"method": "POST"
+		"method": "POST",
+		"urlPathPattern": "/api-messaging/2281/snail-mail",
```
Comment (josc04) [inline src/integration-test/resources/MessageLetterIT/__files/test02_successfully_sendSnailMail/mappings/api-messaging-send-snail-mail-201.json:4]: Add header values for x-origin, x-sent-by and authentication to request match? I.e. add something like:
```
		"headers": {
			"Authorization": {
				"equalTo": "Bearer <dummy-token-value>"
			},
			"x-origin": {
				"equalTo": "PostPortalService"
			},
			"x-sent-by": {
				"equalTo": "joe01doe; type=adAccount"
			}			
		},
```

File: src/test/java/se/sundsvall/postportalservice/integration/messaging/MessagingIntegrationTest.java [hunk 2]
```diff
@@ -96,7 +100,7 @@ void sendDigitalMail() throws SQLException {
 		});
 
 		assertThat(result).isNotNull().isEqualTo(messageBatchResult);
-		verify(messagingClientMock).sendDigitalMail("John Wick; type=adAccount", "Jönssonligan", MUNICIPALITY_ID, digitalMailRequest);
+		verify(messagingClientMock).sendDigitalMail("John Wick; type=adAccount", ORIGIN, MUNICIPALITY_ID, digitalMailRequest);
```
Comment (josc04) [inline src/test/java/se/sundsvall/postportalservice/integration/messaging/MessagingIntegrationTest.java:103]: Create and compare against own variable for origin (with value "PostPortalService") otherwise this test will still go green if someone by accident changes ORIGIN value in contant class, right? Comment goes for all places where constant is used in verifying interactions.

File: src/main/java/se/sundsvall/postportalservice/integration/messaging/MessagingMapper.java [hunk 3]
```diff
@@ -60,4 +63,36 @@ public static DigitalMailAttachment toDigitalMailAttachment(final AttachmentEnti
 			.contentType(DigitalMailAttachment.ContentTypeEnum.fromValue(attachmentEntity.getContentType()));
 	}
 
+	public static SnailmailRequest toSnailmailRequest(final MessageEntity messageEntity, final RecipientEntity recipientEntity) {
```
Comment (josc04) [inline src/main/java/se/sundsvall/postportalservice/integration/messaging/MessagingMapper.java:66]: Make method NPE robust by introducing ofNullable? Also add tests for mapper methods (include verifications on null as inparameter(s)). Comment goes for all mapper methods.

File: src/main/java/se/sundsvall/postportalservice/service/MessageService.java [hunk 4]
```diff
@@ -144,11 +152,16 @@ public String processSmsRequest(final String municipalityId, final SmsRequest sm
 
 	CompletableFuture<Void> processRecipients(final MessageEntity messageEntity) {
 		var futures = messageEntity.getRecipients().stream()
```
Comment (josc04) [inline src/main/java/se/sundsvall/postportalservice/service/MessageService.java:154]: Add ofNullable here to (it is not likely that recipients can be null right now, but it feels nice to have NPE covered if we sometime in the future modifies the logic in the service so that it will return null for recipients at some point)?
